### PR TITLE
test: Add a basic tests for ConnectionNodes component

### PR DESCRIPTION
**Pull request by pdubroy**
- Prepare for a small refactoring for #18904 by introducing some tests for the `ConnectionNodes` component. This is a private component that is an implementation detail of `FilteredConnection`, but is exported in order to have the smallest possible characterization test for this bug.

### CODE REVIEW

File: client/web/src/components/FilteredConnection.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import '@testing-library/jest-dom/extend-expect'
+import { cleanup, render, screen } from '@testing-library/react'
+
+import { ConnectionNodesForTesting as ConnectionNodes } from './FilteredConnection'
+
+const emptyLocation = {
```
Comment (umpox) [inline client/web/src/components/FilteredConnection.test.tsx:7]: Might just be able to do

```
import { useLocation } from 'react-router'
 
const location = useLocation();
```

File: package.json [hunk 2]
```diff
@@ -166,6 +166,7 @@
     "@types/sinon": "9.0.4",
     "@types/socket.io": "2.1.10",
     "@types/socket.io-client": "1.4.33",
+    "@types/testing-library__jest-dom": "^5.9.5",
```
Comment (umpox) [inline package.json:169]: I _think_ that these are already bundled in

File: client/web/src/components/FilteredConnection.test.tsx (hunk 1 above)
Comment (pdubroy) [reply client/web/src/components/FilteredConnection.test.tsx:7]: Where would I put this though? I can't use a hook outside a render function.

Comment (pdubroy) [reply client/web/src/components/FilteredConnection.test.tsx:7]: I could make a helper component that renders the `ConnectionNodes`, though I'm not sure that is necessarily an improvement. WDYT?

File: package.json (hunk 2 above)
Comment (pdubroy) [reply package.json:169]: You're right, I seem to not need this.

File: client/web/src/components/FilteredConnection.test.tsx (hunk 1 above)
Comment (umpox) [reply client/web/src/components/FilteredConnection.test.tsx]: Sorry this was a bad comment by me :D 

```
import { createLocation } from 'history'

const location = createLocation('/')
```

Comment (umpox) [reply client/web/src/components/FilteredConnection.test.tsx]: Getting my routers mixed up!

File: client/web/src/components/FilteredConnection.test.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import '@testing-library/jest-dom/extend-expect'
```
[the remark this answers is not in the document]
Comment (pdubroy) [reply client/web/src/components/FilteredConnection.test.tsx]: Done.

File: client/web/src/components/FilteredConnection.test.tsx (hunk 1 above)
Comment (pdubroy) [reply client/web/src/components/FilteredConnection.test.tsx]: :chef's kiss:

Comment (pdubroy) [reply client/web/src/components/FilteredConnection.test.tsx]: done

**Review by umpox — APPROVED** [review]

**Pull request merged**